solutions/day4.py

```diff
@@ -0,0 +1,82 @@
+
+from utilities import input_reader
+
+xmas = "XMAS"
+count2 = 0
+
+# DIRECTIONS:
+# 1 2 3
+# 4 * 5
+# 6 7 8
+
+def get_next_index(x, y, direction):
+
+    # Based on direction, modify and return x and y
+
+    if direction in [1, 2, 3]:
+        x -= 1
+    elif direction in [6, 7, 8]:
+        x += 1
+    
+    if direction in [1, 4, 6]:
+        y -= 1
+    elif direction in [3, 5, 8]:
+        y += 1
+    
+    return x, y
+
+
+def search(grid, x, y, word, direction):
+
+    # Essentially DFS, but instead of recursively calling for all neighbors,
+    # only call for the index in the 
```

Comment:
the suspense... how is this comment supposed to end?

Reply:
I think I was typing this in class and then my professor started talking about the final😅 I finished typing out what it's supposed to say